oak_runtime/src/node/mod.rs

```diff
@@ -215,17 +258,34 @@ impl NodeFactory<NodeConfiguration> for ServerNodeFactory {
                     .expect("no HTTP configuration provided to Oak Runtime")
                     .http_client_root_tls_certificate
                     .expect("no root TLS certificate provided to Oak Runtime");
-                Ok(Box::new(http::client::HttpClientNode::new(
-                    node_name,
-                    config.clone(),
-                    http_client_root_tls_certificate,
-                )?))
+                Ok(CreatedNode {
+                    instance: Box::new(http::client::HttpClientNode::new(
+                        node_name,
+                        config.clone(),
+                        http_client_root_tls_certificate,
+                    )?),
+                    privilege: http::client::get_privilege(&config.authority),
+                })
             }
             None => Err(ConfigurationError::InvalidNodeConfiguration),
         }
     }
 }
 
+/// A holder struct containing a [`Node`] instance, together with the [`NodePrivilege`] that is
+/// assigned to it.
+///
+/// The reason the privilege is returned here and not by the node itself is that it should be
+/// determined by the [`NodeFactory`] instance, so that an untrusted Node may not be able to
+/// single-handedly increase its own privilege. In this codebase all the Nodes are defined alongside
+/// each other, so this does not make much difference, but if Nodes were loaded dynamically or from
+/// untrusted libraries, then it would be more obvious that we would still want to keep the
+/// privilege assignment here.
+pub struct CreatedNode {
+    pub instance: Box<dyn Node>,
+    pub privilege: NodePrivilege,
```

Comment:
Should this be public? Is the plan to allow modification of the node's privilege at a later time after node creation?

Reply:
There is no such plan, anyways it needs to be passed around, and this makes it more convenient. This may change in a follow up PR though.